entity-framework/core/testing/index.md

```diff
@@ -1,117 +1,26 @@
 
-* Running queries and updates against the same database system used in production.
-* Running queries and updates against some other easier to manage database system.
-* Using test doubles or some other mechanism to avoid using a database at all.
+## Involving the database (or not)
 
-This document outlines the trade-offs involved in each of these choices and shows how EF Core can be used with each approach.
+When writing tests for your EF Core application, one basic decision you need to make is whether your tests will involve your production database system - just as your application does - or whether your tests will run against a [test double](https://martinfowler.com/bliki/TestDouble.html), which replaces your production database system. Two prominent examples of test doubles in the EF Core context are [SQLite in-memory mode](xref:core/testing/choosing-a-testing-strategy#sqlite-as-a-database-fake), and the [InMemory provider](xref:core/testing/choosing-a-testing-strategy#inmemory-as-a-database-fake). For simplicity's sake, we'll use the term *integration tests* to refer to tests which involve your production database systems, and *unit tests* to refer to tests which use a test double.
 
-> [!TIP]
-> See [EF Core testing sample](xref:core/testing/testing-sample) for code demonstrating the concepts introduced here.
+For an in-depth comparison and analysis of integration and unit testing approaches in EF Core applications, see [Choosing a testing strategy](xref:core/testing/choosing-a-testing-strategy). Below is a short point-by-point summary to help you get up to speed with the different options:
 
-## All database providers are not equal
+* Developers frequently turn to unit testing because they believe testing against their production database system is difficult or slow. This isn't always true in our experience, and we suggest giving this approach a chance: [Integration testing](xref:core/testing/integration-testing) provides techniques for reliable, fast integration testing. Writing at least some integration tests is usually necessary in any case - to make sure your application actually works against your production database - and unit testing can be limited in what it allow you to test (see below).
+* The [InMemory provider](xref:core/testing/choosing-a-testing-strategy#inmemory-as-a-database-fake) will not behave like your real database in many important ways. Some features cannot be tested with it at all (e.g. transactions, raw SQL..), some others may behave differently than your production database (e.g. case-sensitivity in queries). While InMemory can work for simple, constrained query scenarios, it is highly limited and we discourage its use.
+  * Mocking `DbSet` for querying suffers from the same disadvantages as the InMemory approach; we discourage this as well.
```

Comment:
```suggestion
  * Mocking `DbSet` for querying is complex and difficult, and suffers from the same disadvantages as the InMemory approach; we discourage this as well.
```

Reply:
OK.

Though if by "mocking DbSet" we mean making DbSet backed by an in-memory collection, and applying LINQ operators as Enumerable ones, then I did this very easily with one of the ready-made EF-specific mocking packages out there (don't remember which one). It really was easy and simple, but doesn't really get you much beyond what InMemory provides.

Otherwise, mocking DbSet could mean creating a whole other LINQ provider (non-EF!), which really doesn't seem worth discussing.

Reply:
What I am saying is that writing a test double that understands/interprets the parameter (i.e. the expression tree) and then returns appropriate test data based on that is very complex/hard. You say this isn't worth discussing, but we have had people try to do just that. (I tried to do it many years ago!)